File: lab_02_Entity_Framework/Supplier.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace JakubBarberEFProducts
{
    
    public class Company
    {
        public int CompanyID { get; set; }
        public String? CompanyName { get; set; }
        public String? Street { get; set; }
        public String? City { get; set; }

        public String? zipCode { get; set; } 
    }
    

    public class Supplier : Company
    {
        public Supplier() { 
            this.Products = new HashSet<Product>();  
        }
        //public int SupplierID { get; set; }
        //public int CompanyID { get; set; }
        //public String? CompanyName { get; set; }
        //public String? Street { get; set; }
        //public String? City { get; set; }
        public long BankAccountNumber { get; set; }

        public ICollection<Product> Products { get; set; }

    }
    
    public class Customer : Company
    {
        //public int CustomerID { get; set; }
        public double Discount { get; set; }
    }
    
}

File: lab_03_Hibernate/code/src/main/java/org/example/Supplier.java
package org.example;

import javax.persistence.*;
import java.util.HashSet;
import java.util.Set;

@Entity
public class Supplier extends Company {
    public Supplier() {
    }

    private String bankAccountNumber;

    public Supplier(String companyName, String street, String city, String zipCode, String banckAccountNumber) {
        super(companyName,street,city,zipCode);
        this.bankAccountNumber = banckAccountNumber;
    }



//    private Adress adress;

    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
//    @JoinColumn(name="supplier_id")
    private Set<Product> productsGroup;

    public Set<Product> getProductsGroup() {
        return productsGroup;
    }

    public void addToProductsGroup(Product product) {
        if (this.productsGroup==null){
            this.productsGroup = new HashSet<Product>();
        }
        this.productsGroup.add(product);
    }
}
